sdk/dotnet/Pulumi.Automation/Events/DiagnosticEvent.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016-2021, Pulumi Corporation
+// NOTE: The class in this file are intended to align with the serialized
+// JSON types defined and versioned in sdk/go/common/apitype/events.go
+
+namespace Pulumi.Automation.Events
+{
+    /// <summary>
+    /// DiagnosticEvent is emitted whenever a diagnostic message is provided, for example errors from
+    /// a cloud resource provider while trying to create or update a resource.
+    /// </summary>
+    public class DiagnosticEvent
+    {
+        public string? Urn { get; }
+
+        public string? Prefix { get; }
+
+        public string Message { get; }
+
+        public string Color { get; }
+
+        public string Severity { get; }
```

Comment:
Perhaps a comment here to match go, or consider an enum:

```
	// Severity is one of "info", "info#err", "warning", or "error".
```

Reply:
I think a comment here should suffice for now. I don't want to make this an enum since it isn't one in any of the other languages and I don't want to back us into a corner in case there's a change. We can consider an enum in the future when we are ready to commit to this set of values.